pipe.py

```diff
@@ -39,8 +39,15 @@ def __init__(self, function):
     def __ror__(self, other):
         return self.function(other)
 
+    def __or__(self, other):
+        return self.__class__(
+            lambda iterable, *args2, **kwargs2: other.function(
+                self.function(iterable, *args2, **kwargs2)
+            )
+        )
```

Comment:
I am just wondering, what if **other** is a `pipeable` object but not an instance of `Pipe` ?!?
Would it not be better to use:

```python
def __or__(self, other):
    cls = self.__class__
    if isinstance(other, Pipe):
        return cls(lambda iterable, *args2, **kwargs2: (
                   other.function(self.function(iterable, *args2, **kwargs2))))

    # -- CASE: other is pipeable only
    return cls(lambda iterable, *args2, **kwargs2: (
               self.function(iterable, *args2, **kwargs2) | other))
```